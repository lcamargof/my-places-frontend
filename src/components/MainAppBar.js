import React from 'react';
import PropTypes from 'prop-types';
import { AppBar, withStyles, Toolbar, Typography } from 'material-ui';
import SearchPlace from './SearchPlace';

const styles = () => ({
  search: { width: 250, float: 'right' },
  flex: { flex: 1 },
});

const MainAppBar = ({ classes, ...props }) => (
  <AppBar position="static" color="default">
    <Toolbar>
      <Typography className={classes.flex} variant="title" color="inherit">
        My Places
      </Typography>
      <SearchPlace {...props} className={classes.search} />
    </Toolbar>
  </AppBar>
);

MainAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
  places: PropTypes.array.isRequired,
  onFilter: PropTypes.func.isRequired,
};

export default withStyles(styles)(MainAppBar);