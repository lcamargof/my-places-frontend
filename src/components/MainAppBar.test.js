import React from 'react';
import { shallow } from 'enzyme';
import { MainAppBar } from './MainAppBar';
import { shallowToJson } from 'enzyme-to-json';

describe('Component: MainAppBar', () => {
  it('Should render correctly', () => {
    const component = shallow(<MainAppBar classes={{}} places={[]} onFilter={jest.fn()} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});