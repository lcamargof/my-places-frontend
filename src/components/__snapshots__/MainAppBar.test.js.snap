// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainAppBar Should render correctly 1`] = `
<WithStyles(AppBar)
  color="default"
  position="static"
>
  <WithStyles(Toolbar)>
    <WithStyles(Typography)
      color="inherit"
      variant="title"
    >
      My Places
    </WithStyles(Typography)>
    <WithStyles(Component)
      onFilter={[Function]}
      places={Array []}
    />
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
